runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsOverlay.java

```diff
@@ -232,6 +231,43 @@ private void renderPlayerOverlay(Graphics2D graphics, Player actor, Color color)
 					textLocation.getY());
 			}
 		}
+
+		if (plugin.isShowAgilityLevel())
+		{
+			if (!plugin.agilityMap.isEmpty() && plugin.agilityMap.keySet().contains(actor.getName()))
+			{
+				if (plugin.getAgilityFormat().equals(PlayerIndicatorsPlugin.AgilityFormats.ICONS))
+				{
+					int width = graphics.getFontMetrics().stringWidth(name);
+					int height = graphics.getFontMetrics().getHeight();
+					if (plugin.agilityMap.get(actor.getName()) >= 70)
+					{
+						OverlayUtil.renderImageLocation(graphics,
+								new Point(textLocation.getX() + 5 + width,
+									textLocation.getY() - height),
+								ImageUtil.resizeImage(agilityIcon, height, height));
```

Comment:
cache this in a `@Getter private BufferedImage agilityScaled = null;` and set it at the top of `if (plugin.isShowAgilityLevel())` using `if (agilityScaled == null) // resize and save`